processing/src/main/java/io/druid/segment/filter/BetweenFilter.java

```diff
@@ -0,0 +1,36 @@
+package io.druid.segment.filter;
+
+import com.google.common.base.Predicate;
+import io.druid.query.filter.BetweenDimFilter;
+
+
+public class BetweenFilter extends DimensionPredicateFilter
+{
+
+  public BetweenFilter(final BetweenDimFilter betweenDimFilter)
+  {
+    super(
+        betweenDimFilter.getDimension(), new Predicate<String>()
+        {
+          @Override
+          public boolean apply(String input)
+          {
+            if (input == null) return false;
+
+            if (betweenDimFilter.isLowerStrict() && betweenDimFilter.isUpperStrict()) {
+              return (input.compareTo(betweenDimFilter.getLower()) < 0) && (input.compareTo(betweenDimFilter.getUpper())
+                                                                            > 0);
+            } else if (betweenDimFilter.isLowerStrict()){
+              return (input.compareTo(betweenDimFilter.getLower()) < 0) && (input.compareTo(betweenDimFilter.getUpper())
+                                                                            >= 0);
+            } else if(betweenDimFilter.isUpperStrict()) {
+              return (input.compareTo(betweenDimFilter.getLower()) <= 0) && (input.compareTo(betweenDimFilter.getUpper())
+                                                                            > 0);
+            }
+            return (input.compareTo(betweenDimFilter.getLower()) <= 0) && (input.compareTo(betweenDimFilter.getUpper())
+                                                                                 >= 0);
+          }
```

Comment:
this is probably going to have very weird results for the `age` use case e.g. this will say that age "100" is smaller than age "20" . I believe, you would need to at least support a flag saying whether the comparison should be done by converting string to numbers.